### PR TITLE
Add `OrderedImports` configuration to disable grouping by import type

### LINKED ISSUE #1128: Feature: Configure sectioning of `orderedImports`
Right now, enabling the `orderedImports` rule will put different types of imports (`@testable`, `@_implementationOnly`, etc.) in their own sections, separated by blank lines.

Would people be opposed if I raised a PR to add a configuration option to disable the sectioning behaviour? Just wanted to get some sense this would be supported by reviewers :)

**My personal rationale**: there are lots of testing scenarios where a list of imports will have one `@testable` import for the module under test and all others as normal imports. In those cases, I thought it seemed like overkill to put the 1 import in its own section. If we don't want to be too prescriptive in `swift-format`, it might be nice to offer the option.

Just let me know if people would be happy with this and I'll work on a PR for it.

### PR DESCRIPTION
**Pull request by amja**
Adds the capability to disable grouping by import type (`@testable`, `@_implementationOnly`, etc.) and only order the imports lexicographically.

This is achieved via a configuration option `orderedImports.shouldGroupImportTypes`.

Updated testing and documentation to reflect this new option.

Closes #1128

### CODE REVIEW

File: Sources/SwiftFormat/API/Configuration.swift [hunk 1]
```diff
@@ -563,7 +563,8 @@ public struct NoAssignmentInExpressionsConfiguration: Codable, Equatable {
 /// Configuration for the `OrderedImports` rule.
 public struct OrderedImportsConfiguration: Codable, Equatable {
   /// Determines whether imports within conditional compilation blocks should be ordered.
-  public var includeConditionalImports: Bool = false
-
+  public var includeConditionalImports = false
+  /// Determines whether imports are separated into groups based on their type.
+  public var shouldGroupImportTypes = true
```
Comment (allevato) [inline Sources/SwiftFormat/API/Configuration.swift]:
```suggestion
  public var shouldGroupImports = true
```

I don't think the word "types" adds anything here (and could be confusable with what "types" means otherwise in Swift).

File: Sources/SwiftFormat/Rules/OrderedImports.swift [hunk 2]
```diff
@@ -39,16 +41,19 @@ public final class OrderedImports: SyntaxFormatRule {
...
-    var codeBlocks: [Line] = []
-    var fileHeader: [Line] = []
+    var formattedLines = [Line]()
+
+    var regularImports = [Line]()
+    var declImports = [Line]()
+    var implementationOnlyImports = [Line]()
+    var testableImports = [Line]()
+    // Used if `shouldGroupImportTypes` configuration is `false`.
+    var combinedImports = [Line]()
```
[the remark this answers is not in the document]
Comment (amja) [reply Sources/SwiftFormat/Rules/OrderedImports.swift]: I added it mainly for clarity so that the name `regularImports` isn't misleading when the new config is enabled. But it works fine putting everything in `regularImports` so I've updated to do that instead.

File: Tests/SwiftFormatTests/Rules/OrderedImportsTests.swift [hunk 3]
```diff
@@ -1018,4 +1018,71 @@ final class OrderedImportsTests: LintOrFormatRuleTestCase {
...
+      OrderedImports.self,
+      input: code,
+      expected: code,
+      findings: [
+      ],
+      configuration: configuration
+    )
+  }
+
+  func testInvalidOrderingWithGroupingDisabled() {
```
[the remark this answers is not in the document]
Comment (amja) [reply Tests/SwiftFormatTests/Rules/OrderedImportsTests.swift]: Thanks for this suggestion. Adding the test uncovered a couple of issues that I have also resolved!

If you have any more thoughts for edge cases I should test, please let me know :)
